Tests/powershell-yaml.Tests.ps1

```diff
@@ -92,6 +92,31 @@ InModuleScope $moduleName {
             }
         }
 
+        Context "Nulls and strings" {
+            $nullAndString = @{"iAmNull"= $null; "iAmEmptyString"=""}
+            $yaml = "iAmNull: 
```

Comment:
This causes an error on Windows 2012 R2 (Powershell 4):

Try this: https://paste.ubuntu.com/p/FjTBPxhRp5/

Reply:
Notice the extra newline at the end, in the multi-line text.

Reply:
I wasn't able to reproduce that on a new Azure VM 😕 
```
PS C:\Users\chrisv\Documents\WindowsPowerShell\Modules\powershell-yaml\Tests> $psversiontable

Name                           Value
----                           -----
PSVersion                      4.0
WSManStackVersion              3.0
SerializationVersion           1.1.0.1
CLRVersion                     4.0.30319.42000
BuildVersion                   6.3.9600.19170
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}
PSRemotingProtocolVersion      2.2

PS C:\Users\chrisv\Documents\WindowsPowerShell\Modules\powershell-yaml\Tests> powershell -command {invoke-pester}
Executing all tests in '.'

Executing script C:\Users\chrisv\Documents\WindowsPowerShell\Modules\powershell-yaml\Tests\powershell-yaml.Tests.ps1

  Describing Test encode-decode symmetry.
...
    Context Nulls and strings
      [+] should preserve nulls and empty strings from PowerShell 133ms
      [+] should preserve nulls and empty strings from Yaml 36ms
```

Reply:
Ooh.  Does your tests file have Unix or Windows line endings?

Reply:
I've updated this to use a `StringBuilder` to construct the expected yaml.  This should insert the appropriate line endings, regardless of platform.  Tested on 4, 5 and 6/windows, just waiting for a `do-release-upgrade` to test on 6/linux.

Reply:
if you just construct the yaml like:

```powershell
# This is how you devine a multi-line string in PowerShell
# Note the extra empty line at the end.
$yaml = @"
iAmNull: 
iAmEmptyString: ""

"@
```
it will work everywhere. No need to fiddle with ```Replace()```.

Reply:
When I do that, I get:
```
        Expected strings to be the same, but they were different.
        Expected length: 29
        Actual length:   31
        Strings differ at index 9.
        Expected: 'iAmNull:
        iAmEmptyString: ""
        '
        But was:  'iAmNull: \r
        iAmEmptyString: ""\r
        '
        --------------------^
        109:                 $toYaml | Should Be $yaml
```

Reply:
between @" and "@ you should have 3 lines:

iAmNull: 
iAmEmptyString: ""

and an empty line.

Reply:
This is my result on linux:

https://pastebin.com/Bn02nfkE

Reply:
Notice how YamlDotNet adds an extra newline at the end of the rendered yaml:

```powershell
PS /tmp/powershell-yaml> Import-Module ./powershell-yaml.psd1
PS /tmp/powershell-yaml> $a = @{"a"=1; "b"=2}
PS /tmp/powershell-yaml> 
PS /tmp/powershell-yaml> ConvertTo-Yaml $a
a: 1
b: 2

PS /tmp/powershell-yaml> 
```

There is a space between ```b: 2``` and the next prompt.

Reply:
Apologies. I should probably stop reading code at 2 AM. Just noticed that the expected output does not include that empty line.

What system are you running on? And would you mind running the tests using:

```
pwsh -NonInteractive -Command Invoke-Pester
```

On ubuntu it works as expected.

Reply:
> EDIT: The error may have been cause by git. On Windows, git may check out code with CRLF, but commits it LF. The above multi line definition works on all windows versions I tested, and linux.

ahh yes, I have `core.autocrlf=false` set in git config so I'm getting LF everywhere.

Reply:
Change applied

Reply:
> What system are you running on?

Win10 with Powershell 5 and 6, plus 6 in WSL, and a 2012 R2 VM up on Azure.

All tests were run with `powershell -command {invoke-pester}` or `pwsh -command invoke-pester`

Reply:
awesome! Thanks!

Reply:
Yeah it passes if I save the file with line endings to match the OS the tests are running on.  Stupid line endings.